lib/PhantomEnvironment.js

```diff
@@ -157,151 +221,134 @@ class PhantomEnvironment extends AbstractEnvironment {
-              }, 500);
-
-              return deferred.promise();
-            });
-  }
-
-  waitForPage(timeout) {
-    timeout = timeout || 5000;
-
-    const deferred = vow.defer();
-    const timeoutId = setTimeout(() => {
-      debug('Timeout %s has reached on page load', timeout);
-      this._navigationActions = [];
-      deferred.reject(new Error('Page navigation timeout'));
-    }, timeout);
-
-    this._navigationActions.push((err) => {
-      clearTimeout(timeoutId);
-      if (err) {
-        deferred.reject(err);
-      } else {
-        deferred.resolve();
-      }
-    });
-    debug('Added page load callback');
+    await mkdir(screenShotFilePath);
+
+    const windowSize = {
+      left: 0,
+      top: 0,
+      width: options.screen.width,
+      height: options.screen.height,
+    };
+    this._page.clipRect = windowSize;
+    debug('Doing snapshot with window size %o, filepath %s', windowSize, screenShotFileName);
+    this._page.render(screenShotFileName);
+
+    await new Promise((resolve, reject) => {
+      let timeout;
+
+      const interval = setInterval(() => {
+        if (fs.statSync(screenShotFilePath).size) {
+          clearInterval(interval);
+          clearTimeout(timeout);
+          resolve();
+        }
+      }, 20);
 
-    return deferred.promise().then(this._injectVendors, this);
+      timeout = setTimeout(() => {
+        clearInterval(interval);
+        reject(new Error('Snapshot timeout'));
+      }, 500);
+    });
   }
 
-  waitForQuery(uri, timeout) {
-    timeout = timeout || 5000;
-
-    const deferred = vow.defer();
-    const timeoutId = setTimeout(() => {
-      debug('Timeout %s has reached for waiting query %s', timeout, uri);
-      this._requestingActions = [];
-      deferred.reject(new Error('Waiting request timeout'));
-    }, timeout);
+  async waitForPage(timeout = 5000) {
+    await new Promise((resolve, reject) => {
+      const timeoutId = setTimeout(() => {
+        debug('Timeout %s has reached on page load', timeout);
+        this._navigationActions = [];
+        reject(new Error('Page navigation timeout'));
+      }, timeout);
 
-    this._requestingActions.push({
-      pattern: uri,
-      fn(err, results) {
+      this._navigationActions.push((err) => {
         clearTimeout(timeoutId);
         if (err) {
-          deferred.reject(err);
+          reject(err);
         } else {
-          deferred.resolve(results);
+          resolve();
         }
-      },
+      });
+      debug('Added page load callback');
+    });
+    await this._injectVendors();
+  }
+
+  waitForQuery(uri, timeout = 5000) {
+    return new Promise((resolve, reject) => {
+      const timeoutId = setTimeout(() => {
+        debug('Timeout %s has reached for waiting query %s', timeout, uri);
+        this._requestingActions = [];
+        reject(new Error('Waiting request timeout'));
+      }, timeout);
+
+      this._requestingActions.push({
+        pattern: uri,
+        fn(err, results) {
```

Comment:
Just in case you can clarify this line: don't see any `fn` defined in surrounding (in the old code as well)

Reply:
But it's the declaration itself. Strange that it's not an arrow function. But still PR is not about refactoring everything, so I would prefer to fix it in next PRs.

```js
const a = {
  fn: function() {}
}
```

is the same as 

```js
const a = {
  fn() {}
};
```